src/router/storeSync/LayerParamConfig.class.js

```diff
@@ -138,20 +141,18 @@ function dispatchLayersFromUrlIntoStore(store, urlParamValue) {
         }
     })
     // adding any layer that is not present yet
-    parsedLayers.forEach((layer) => {
+    parsedLayers.forEach((parsedLayer) => {
         if (
-            !store.state.layers.activeLayers.find((activeLayer) => activeLayer.getID() === layer.id)
+            !store.state.layers.activeLayers.find(
+                (activeLayer) => activeLayer.getID() === parsedLayer.id
+            )
         ) {
-            // checking if it is an external layer first
-            const externalLayer = transformParsedExternalLayerIntoObject(layer)
-            if (externalLayer) {
-                log.debug(`  Add external layer ${layer.id} to active layers`, externalLayer)
-                promisesForAllDispatch.push(store.dispatch('addLayer', externalLayer))
-            } else {
-                // if internal (or BOD) layer, we add it through its parsed config
-                log.debug(`  Add layer ${layer.id} to active layers`, layer)
-                promisesForAllDispatch.push(store.dispatch('addLayer', layer))
+            const layerObject = createLayerObject(parsedLayer)
+            if (layerObject.type == LayerTypes.KML && layerObject.adminId) {
+                promisesForAllDispatch.push(store.dispatch('setOpenOnAdminId', true))
             }
+            log.debug(`  Add layer ${parsedLayer.id} to active layers`, layerObject)
+            promisesForAllDispatch.push(store.dispatch('addLayer', layerObject))
```

Comment:
Isn't there here a potential race condition? In the drawing module, you watch only for the active KML layer, and here you run "setOpenOnAdminId" and "addLyer" concurrently. So if the latter action finishes before the former, the drawing mode would not be opened I think.

Reply:
Good point. It works because for now both event occurs before the drawing module is created and that in created event of the module I call the `openDrawingOnAdminId` method. But for safety I'll watch also `setOpenOnAdminId` this would also cover the case where we would add the adminId manually in the url afterwards.

Reply:
Done